hack/generator/pkg/astmodel/package_reference.go

```diff
@@ -18,12 +20,26 @@ const (
 // a struct belongs to.
 type PackageReference struct {
 	packagePath string
+	name        *string
```

Comment:
hmm, there's not much sense in having a `name` on a `PackageReference` when it is part of a `Type`. do we want to split this into `PackageName` (to use on a `Type`) and a `PackageImport` (= `PackageName` + friendly name) to be used for imports?

Reply:
Yeah, agree -- I thought about doing something like this but wasn't 100% sure. Can fix.

Reply:
I left this one called PackageReference, and made a new one PackageImport.